host/client_integration_test.go

```diff
@@ -442,6 +452,119 @@ func (s *clientIntegrationSuite) TestClientDataConverter_WithChild() {
 	s.Equal(2, d.NumOfCallFromPayloads)
 }
 
+func (s *clientIntegrationSuite) TestTooManyChildWorkflows() {
+	// To ensure that there is one pending child workflow before we try to create the next one,
+	// we create a child workflow here that signals the parent when it has started and then blocks forever.
+	parentWorkflowId := "client-integration-too-many-child-workflows"
+	blockingChildWorkflow := func(ctx workflow.Context) error {
+		workflow.SignalExternalWorkflow(ctx, parentWorkflowId, "", "blocking-child-started", nil)
+		workflow.GetSignalChannel(ctx, "unblock-child").Receive(ctx, nil)
+		return nil
+	}
+	childWorkflow := func(ctx workflow.Context) error {
+		return nil
+	}
+
+	maxPendingChildWorkflows := s.maxPendingChildExecutions
+	parentWorkflow := func(ctx workflow.Context) error {
+		childStarted := workflow.GetSignalChannel(ctx, "blocking-child-started")
+		for i := 0; i < maxPendingChildWorkflows; i++ {
```

Comment:
just do maxPendingChildWorkflows + 1?

Reply:
I make the last child non-blocking so that it will complete as soon as one of the other child workflows is unblocked. If I made it also blocking, it would be unclear what caused the parent workflow to finally complete.